nui/src/main/java/org/terasology/nui/widgets/types/math/RectanglefWidgetFactory.java

```diff
@@ -0,0 +1,98 @@
+                            Rectanglef old = rectBinding.get();
+                            rectBinding.set(RectUtility.createFromMinAndSize(value, old.minY, old.maxX - old.minX, old.maxY - old.minY));
+                        }
+                    })
+                .add("y",
+                    rectBinding -> new Binding<Float>() {
+                        @Override
+                        public Float get() {
+                            return rectBinding.get().minY;
+                        }
+
+                        @Override
+                        public void set(Float value) {
+                            Rectanglef old = rectBinding.get();
+                            rectBinding.set(RectUtility.createFromMinAndSize(old.minX, value, old.maxX - old.minX, old.maxY - old.minY));
+                        }
+                    })
+                .add("w",
+                    rectBinding -> new Binding<Float>() {
+                        @Override
+                        public Float get() {
+                            Rectanglef rect = rectBinding.get();
+                            return rect.maxX - rect.minX;
+                        }
+
+                        @Override
+                        public void set(Float value) {
+                            Rectanglef old = rectBinding.get();
+                            rectBinding.set(RectUtility.createFromMinAndSize(old.minX, old.minY, old.minX + value, old.maxY - old.minY));
+                        }
+                    })
+                .add("h",
+                    rectBinding -> new Binding<Float>() {
+                        @Override
+                        public Float get() {
+                            Rectanglef rect = rectBinding.get();
+                            return rect.maxY - rect.minY;
+                        }
+
+                        @Override
+                        public void set(Float value) {
+                            Rectanglef old = rectBinding.get();
+                            rectBinding.set(RectUtility.createFromMinAndSize(old.minX, old.minY, old.maxX - old.minX, old.minY + value));
+                        }
+                    });
+
+        return Optional.of((TypeWidgetBuilder<T>) builder);
+    }
+
+    private static class RectanglefWidgetBuilder extends LabeledNumberFieldRowBuilder<Rectanglef, Float> {
+        public RectanglefWidgetBuilder(TypeWidgetLibrary library) {
+            super(Rectanglef.class, float.class, library);
+        }
+
+        @Override
+        protected Rectanglef getDefaultValue() {
+            // Make non-empty so that editing works as intended
+            // When the initial rect is empty, editing any of the components will make no difference
+            // since one of the size components will always be zero, making the factory methods always
+            // return the empty rect
+            return RectUtility.createFromMinAndSize(0f, 0f, 1f, 1f);
```

Comment:
**request:** This is now longer the empty rectangle, at least not with the current implementation, I think... see https://github.com/MovingBlocks/joml-ext/pull/23

You could create an "empty" rectangle of size 0 around the point (0,0) by simply doing
```suggestion
            return new Rectanglef(0, 0);
```